ecs-cli/modules/cli/local/up_app.go

```diff
@@ -14,18 +14,182 @@
 package local
 
 import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/local/converter"
 	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/local/docker"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/local/localproject"
 	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/local/network"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/local/secrets"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands/flags"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	composeV3 "github.com/docker/cli/cli/compose/types"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 	"github.com/urfave/cli"
 )
 
+type ssmDecrypter struct {
+	client *ssm.SSM
+}
+
+type secretsManagerDecrypter struct {
+	client *secretsmanager.SecretsManager
+}
```

Comment:
hmmmm what do you think about having these client objects live somewhere else? I feel like the <command>_app.go files should only contain "controller" logic rather than "model" logic (if we're thinking about this from a MVC sort of perspective). It feels sort of weird to have these and their `DecryptMethod` functions defined here.

Also, the fact that this is duplicated maybe hints at having a secretsDecrypter interface that can be implemented by either SSM or SecretsManager.

Reply:
Ah, got it. Any reason not to put the interface in the same place as the implementation? (non-blocking)

Reply:
Just to follow the guidelines:
> Go interfaces generally belong in the package that uses values of the interface type, not the package that implements those values. The implementing package should return concrete (usually pointer or struct) types: that way, new methods can be added to implementations without requiring extensive refactoring.

https://github.com/golang/go/wiki/CodeReviewComments#interfaces